From 276c5f05a997bc9326b5440e64538fefa6899151 Mon Sep 17 00:00:00 2001
From: Ricardo Constantino <wiiaboo@gmail.com>
Date: Thu, 24 Aug 2017 19:27:35 +0100
Subject: [PATCH] fix fc-blank.h generation if unicode.org is disabled

---
 fc-blanks/fc-blanks.py | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/fc-blanks/fc-blanks.py b/fc-blanks/fc-blanks.py
index cc23cde..5ad7e91 100755
--- a/fc-blanks/fc-blanks.py
+++ b/fc-blanks/fc-blanks.py
@@ -11,28 +11,32 @@ except ImportError:
 
 import sys
 import os
-from lxml import html
+from lxml.html import fromstring
 from six.moves import range
 
 datafile = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'list-unicodeset.html')
+dom = None
 try:
     fp = urlopen('http://unicode.org/cldr/utility/list-unicodeset.jsp?a=[%3AGC%3DZs%3A][%3ADI%3A]&abb=on&ucd=on&esc=on&g')
     data = fp.read()
     fp.close()
+    dom = fromstring(data)
+    if not dom.xpath('/html/body/form/p/text()'):
+        raise URLError("Wrong content.")
     fp = open(datafile, 'wb');
     fp.write(data);
     fp.close();
 except URLError:
     # fall back reading the static data in repo
     try:
-        fp = open(datafile)
+        fp = open(datafile, 'rb')
         data = fp.read()
         fp.close()
+        dom = fromstring(data)
     except IOError:
         sys.stderr.write("Error: No static data to generate the blank data. please make sure the network connection is reachable to Unicode.org\n")
         sys.exit(1)
 
-dom = html.fromstring(data)
 x = dom.xpath('/html/body/form/p/text()')
 p = x[1]
 if p[0] == '[' and p[-1] == ']':
-- 
2.13.3

